analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -129,7 +129,7 @@ or CompilationUnitSyntax
                 _ => Enumerable.Empty<string>()
             };
 
-            return parameterList.Union(ParentParameterList(node));
+            return parameterList.Union(ParentParameterList(creationSyntax));
 
             static IEnumerable<string> IdentifierNames(BaseParameterListSyntax parameterList) =>
                     parameterList.Parameters.Select(x => x.Identifier.ValueText);
```

Comment:
Do we have a test case where the declared name does not match the reference, like here?

```cs
public class C(int @a) {
    public void M() {
        a = 1;
    }
}
```

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQEkB2AbAlpiAQQQSgE9kAaAExAGoAfAAQGYACJgJjYGEAKfDA5MoASjYBvALAAoNvI7smAFjYBZPuOlyFuqGwC8bAIwBuWboC+sy0A==

Reply:
We don't have a test case but it's supported (no FP/FN), I added a few examples.